core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessage.java

```diff
@@ -149,6 +152,16 @@ private void subscribe(SubscriptionImpl subscription, Subscriber<Object> subscri
         }
     }
 
+    /**
+     * Invoked when a subscriber subscribes.
+     */
+    protected void subscribe0(EventExecutor executor, boolean withPooledObjects, boolean notifyCancellation) {}
```

Comment:
> Shouldn't we use SubscriptionOption... in case another option is added later?

That's a good point. Let me refator `AbstractStreamMessage` a bit.

Reply:
> Maybe onSubscribe? to align with onRequest

Did You mean to rename `subscribe0` to `onSubscribe`? As you know `onSubscribe` is an API of `Subscriber`. I think if we use `onSubscribe` here, it could be a misterm and cause confusion a bit.

Reply:
That is true. 👍